File: 2D PVP Action Game Server/Packet/Buffer_Converter.cpp
#define _CRT_SECURE_NO_WARNINGS
#include "Buffer_Converter.h"
#include <string.h>

Buffer_Converter::Buffer_Converter() {

}
Buffer_Converter::~Buffer_Converter() {

}

HeaderType Buffer_Converter::GetHeader(char* buffer)
{
	HeaderType header;
	memcpy(&header, buffer + 4, sizeof(HeaderType));

	return header;
}

File: Assets/Script/Server/Buffer_Converter.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.InteropServices;
using UnityEngine;

public class Buffer_Converter : MonoBehaviour
{
    public int GetHeaderType(byte[] buffer)
    {
        byte[] dataHeaderType = new byte[Marshal.SizeOf(typeof(ServerUtil.Header.HeaderType))];

        Buffer.BlockCopy(buffer, Marshal.SizeOf(typeof(int)), dataHeaderType, 0, dataHeaderType.Length);

        return Convert.ToInt32(dataHeaderType);
    }
}
